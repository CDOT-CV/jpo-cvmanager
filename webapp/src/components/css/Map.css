.rsu-marker {
    margin-top: -6px;
    margin-left: -6px;
}

.legend {
    z-index: 90;
    background: #0e2052;
    margin: 10px;
    border-radius: 15px;
}

.legend-header {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 24px;
    margin-top: 5px;
    margin-left: 10px;
    margin-right: 10px;
    color: white;
}

.legend-item {
    margin-top: 8px;
    margin-bottom: 8px;
    margin-left: 10px;
    margin-right: 10px;
}

.legend-label {
    font-family: Arial, Helvetica, sans-serif;
    font-size: 18px;
    color: white;
}

.legend-input {
    margin-right: 10px;
    accent-color: #d16d15;
}
