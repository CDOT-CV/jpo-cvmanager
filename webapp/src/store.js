import { configureStore } from "@reduxjs/toolkit";
import rsuReducer from "./slices/rsuSlice";
import userReducer from "./slices/userSlice";
import wzdxReducer from "./slices/wzdxSlice";
import configReducer from "./slices/configSlice";

const store = configureStore({
  reducer: {
    rsu: rsuReducer,
    user: userReducer,
    wzdx: wzdxReducer,
    config: configReducer,
  },
});

export default store;
