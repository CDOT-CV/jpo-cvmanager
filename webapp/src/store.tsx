import { Action, ThunkAction, configureStore } from '@reduxjs/toolkit'
import rsuReducer from './generalSlices/rsuSlice'
import userReducer from './generalSlices/userSlice'
import wzdxReducer from './generalSlices/wzdxSlice'
import configReducer from './generalSlices/configSlice'
import adminAddOrganizationReducer from './features/adminAddOrganization/adminAddOrganizationSlice'
import adminAddRsuReducer from './features/adminAddRsu/adminAddRsuSlice'
import adminAddUserReducer from './features/adminAdduser/adminAddUserSlice'
import adminEditOrganizationReducer from './features/adminEditOrganization/adminEditOrganizationSlice'
import adminEditRsuReducer from './features/adminEditRsu/adminEditRsuSlice'
import adminEditUserReducer from './features/adminEditUser/adminEditUserSlice'
import adminOrganizationTabReducer from './features/adminOrganizationTab/adminOrganizationTabSlice'
import adminOrganizationTabUserReducer from './features/adminOrganizationTabUser/adminOrganizationTabUserSlice'
import adminOrganizationTabRsuReducer from './features/adminOrganizationTabRsu/adminOrganizationTabRsuSlice'
import adminRsuTabReducer from './features/adminRsuTab/adminRsuTabSlice'
import adminUserTabReducer from './features/adminUserTab/adminUserTabSlice'
import menuReducer from './features/menu/menuSlice'
import intersectionMapReducer from './components/map/map-slice'
import intersectionMapLayerStyleReducer from './components/map/map-layer-style-slice'

export const setupStore = (preloadedState: any) => {
  return configureStore({
    reducer: {
      rsu: rsuReducer,
      user: userReducer,
      wzdx: wzdxReducer,
      config: configReducer,
      adminAddOrganization: adminAddOrganizationReducer,
      adminAddRsu: adminAddRsuReducer,
      adminAddUser: adminAddUserReducer,
      adminEditOrganization: adminEditOrganizationReducer,
      adminEditRsu: adminEditRsuReducer,
      adminEditUser: adminEditUserReducer,
      adminOrganizationTab: adminOrganizationTabReducer,
      adminOrganizationTabUser: adminOrganizationTabUserReducer,
      adminOrganizationTabRsu: adminOrganizationTabRsuReducer,
      adminRsuTab: adminRsuTabReducer,
      adminUserTab: adminUserTabReducer,
      menu: menuReducer,
      intersectionMap: intersectionMapReducer,
      intersectionMapLayerStyle: intersectionMapLayerStyleReducer,
    },
    preloadedState,
  })
}

type AppStore = ReturnType<typeof setupStore>
export type AppState = ReturnType<AppStore['getState']>
type AppThunk<ReturnType = void> = ThunkAction<ReturnType, AppState, unknown, Action>

export type RootState = ReturnType<ReturnType<typeof setupStore>['getState']>
