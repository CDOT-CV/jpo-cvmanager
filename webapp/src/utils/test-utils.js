/**
 * This function searches for the every react-aria SSR ids in a given HTMLElement node and replace every attribute values with a static id
 *
 * This can be usefull when you're trying to generate a snapshot of components using react-aria under the hood
 *
 * @ex :
 * ```
 * const { container } = render(<Component />);
 *
 * replaceReactAriaIds(container);
 * ```
 *
 * @param container The HTMLElement node to search for SSR ids
 */
function replaceChaoticIds(container) {
  const props = [
    {
      selector: "class",
      updateFunc: (val) =>
        val.replace(
          /css-[0-9a-z]{6}-MuiTableCell-root-MuiTablePagination-root/g,
          "css-mocked-MuiTableCell-root-MuiTablePagination-root"
        ),
    },
  ];

  container.querySelectorAll("td").forEach((item) => {
    props.forEach((prop) => {
      if (item.getAttribute(prop.selector)) {
        item.setAttribute(prop.selector, prop.updateFunc(item.getAttribute(prop.selector)));
      }
    });
  });
  container.querySelectorAll("input[aria-invalid]").forEach((item) => {
    delete item["aria-invalid"];
  });
  return container;
}

export { replaceChaoticIds };
