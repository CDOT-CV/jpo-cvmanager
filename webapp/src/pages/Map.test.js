import React from 'react'
import { render, screen, fireEvent, queryByAttribute } from '@testing-library/react'
import Map from './Map'
import { Provider } from 'react-redux'
import { setupStore } from '../store'
import { replaceChaoticIds } from '../utils/test-utils'

it('snapshot msgCoordinates wzdx', () => {
  const initialState = {
    rsu: {
      value: {
        msgCoordinates: [],
        rsuCounts: {},
        mapList: [],
        msgData: [],
        msgStart: '2023-05-10T03:24:00',
        msgEnd: '2023-05-10T03:25:00',
        msgCoordinates: [
          [-104.9903, 39.7392],
          [-104.9904, 39.7393],
          [-104.9905, 39.7391],
        ],
      },
    },
    wzdx: {
      value: {
        features: [
          {
            properties: {
              core_details: {
                road_names: ['road_name_1'],
                direction: 'direction',
                description: 'description',
              },
              vehicle_impact: 'vehicle_impact',
              workers_present: 'workers_present',
              start_date: 'start_date',
              end_date: 'end_date',
            },
            geometry: {
              coordinates: [
                [-104.9903, 39.7392],
                [-104.9908, 39.7398],
                [-104.9909, 39.7399],
              ],
            },
          },
        ],
      },
    },
  }
  const { container } = render(
    <Provider store={setupStore(initialState)}>
      <Map />
    </Provider>
  )

  fireEvent.click(screen.queryByText('RSU'))
  fireEvent.click(screen.queryByText('Heatmap'))
  fireEvent.click(screen.queryByText('Message Viewer'))
  // fireEvent.click(screen.queryByText('WZDx'))

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})

it('snapshot message viewer clicked', () => {
  const initialState = {
    rsu: {
      value: {
        msgCoordinates: [],
        rsuCounts: {},
        mapList: [],
        msgStart: '2023-05-10T03:24:00',
        msgFilterStep: 60, // 1 hour
        msgFilterOffset: 24 * 4, // 4 days
        msgData: [
          {
            properties: {
              time: new Date('2023-05-12T03:24:00'),
            },
          },
          {
            properties: {
              time: new Date('2023-05-17T03:24:00'),
            },
          },
        ],
      },
    },
    config: {
      value: {
        configCoordinates: [],
        configList: ['1.1.1.1', '2.2.2.2', '3.3.3.3'],
        addConfigPoint: false,
      },
    },
  }
  const { container } = render(
    <Provider store={setupStore(initialState)}>
      <Map />
    </Provider>
  )

  expect(replaceChaoticIds(container)).toMatchSnapshot()
})
