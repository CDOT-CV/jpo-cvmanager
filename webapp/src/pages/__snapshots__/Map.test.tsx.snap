// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot bsmCoordinates wzdx 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="MuiGrid-root MuiGrid-container legend-grid css-mocked-MuiGrid-root"
    >
      <div
        class="legend"
      >
        <h1
          class="legend-header"
        >
          Map Layers
        </h1>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              checked=""
              class="legend-input"
              type="checkbox"
            />
            RSU
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              checked=""
              class="legend-input"
              type="checkbox"
            />
            Heatmap
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              checked=""
              class="legend-input"
              type="checkbox"
            />
            BSM Viewer
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              checked=""
              class="legend-input"
              type="checkbox"
            />
            WZDx
          </label>
        </div>
      </div>
      <div
        class="rsu-status-div"
      >
        <h1
          class="legend-header"
        >
          RSU Status
        </h1>
        <label
          class="rsu-status-label"
        >
          <input
            checked=""
            class="rsu-status-input"
            name="none-status-radio"
            type="radio"
            value="none"
          />
          None
        </label>
        <label
          class="rsu-status-label"
        >
          <input
            class="rsu-status-input"
            name="online-status-radio"
            type="radio"
            value="online"
          />
          Online Status
        </label>
        <label
          class="rsu-status-label"
        >
          <input
            class="rsu-status-input"
            name="scms-status-radio"
            type="radio"
            value="scms"
          />
          SCMS Status
        </label>
        <h1
          class="legend-header"
        >
          RSU Configuration
        </h1>
        <div
          class="MuiFormGroup-root MuiFormGroup-row form-group-row css-mocked-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mocked-MuiFormControlLabel-root"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-mocked-MuiSwitch-root"
            >
              <span
                class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-mocked-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                  type="checkbox"
                />
                <span
                  class="MuiSwitch-thumb css-mocked-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-mocked-MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mocked-MuiTypography-root"
            >
              Add Points
            </span>
          </label>
        </div>
        <div
          class="MuiFormGroup-root MuiFormGroup-row css-mocked-MuiFormGroup-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium contained-button css-mocked-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Configure RSUs
          </button>
        </div>
      </div>
    </div>
    <div
      class="container-fluid"
      style="width: 100%; height: calc(100vh - 100px); display: flex;"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      />
    </div>
    <div
      class="control"
    >
      <div
        class="buttonContainer"
      >
        <button
          class="button"
        >
          Add Point
        </button>
        <button
          class="button"
        >
          Clear
        </button>
      </div>
      <div
        class="dateContainer"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-mocked-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-mocked-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-1"
            id="mui-1-label"
          >
            Select start date
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-mocked-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-mocked-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-1"
              placeholder="mm/dd/yyyy hh:mm (a|p)m"
              type="tel"
              value="05/10/2023 03:24 AM"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-mocked-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is May 10, 2023"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-mocked-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mocked-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Select start date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="dateContainer"
      >
        <div
          class="MuiFormControl-root MuiTextField-root css-mocked-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-mocked-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="mui-2"
            id="mui-2-label"
          >
            Select end date
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-mocked-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-mocked-MuiInputBase-input-MuiOutlinedInput-input"
              id="mui-2"
              placeholder="mm/dd/yyyy hh:mm (a|p)m"
              type="tel"
              value="05/10/2023 03:25 AM"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-mocked-MuiInputAdornment-root"
            >
              <button
                aria-label="Choose date, selected date is May 10, 2023"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-mocked-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-mocked-MuiSvgIcon-root"
                  data-testid="CalendarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-mocked-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  Select end date
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="submitContainer"
      >
        <button
          id="submitButton"
        >
          Submit
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot bsmData clicked 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="MuiGrid-root MuiGrid-container legend-grid css-mocked-MuiGrid-root"
    >
      <div
        class="legend"
      >
        <h1
          class="legend-header"
        >
          Map Layers
        </h1>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              checked=""
              class="legend-input"
              type="checkbox"
            />
            RSU
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              class="legend-input"
              type="checkbox"
            />
            Heatmap
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              class="legend-input"
              type="checkbox"
            />
            BSM Viewer
          </label>
        </div>
        <div
          class="legend-item"
        >
          <label
            class="legend-label"
          >
            <input
              class="legend-input"
              type="checkbox"
            />
            WZDx
          </label>
        </div>
      </div>
      <div
        class="rsu-status-div"
      >
        <h1
          class="legend-header"
        >
          RSU Status
        </h1>
        <label
          class="rsu-status-label"
        >
          <input
            checked=""
            class="rsu-status-input"
            name="none-status-radio"
            type="radio"
            value="none"
          />
          None
        </label>
        <label
          class="rsu-status-label"
        >
          <input
            class="rsu-status-input"
            name="online-status-radio"
            type="radio"
            value="online"
          />
          Online Status
        </label>
        <label
          class="rsu-status-label"
        >
          <input
            class="rsu-status-input"
            name="scms-status-radio"
            type="radio"
            value="scms"
          />
          SCMS Status
        </label>
        <h1
          class="legend-header"
        >
          RSU Configuration
        </h1>
        <div
          class="MuiFormGroup-root MuiFormGroup-row form-group-row css-mocked-MuiFormGroup-root"
        >
          <label
            class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-mocked-MuiFormControlLabel-root"
          >
            <span
              class="MuiSwitch-root MuiSwitch-sizeMedium css-mocked-MuiSwitch-root"
            >
              <span
                class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary css-mocked-MuiButtonBase-root-MuiSwitch-switchBase"
              >
                <input
                  class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                  type="checkbox"
                />
                <span
                  class="MuiSwitch-thumb css-mocked-MuiSwitch-thumb"
                />
                <span
                  class="MuiTouchRipple-root css-mocked-MuiTouchRipple-root"
                />
              </span>
              <span
                class="MuiSwitch-track css-mocked-MuiSwitch-track"
              />
            </span>
            <span
              class="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-mocked-MuiTypography-root"
            >
              Add Points
            </span>
          </label>
        </div>
        <div
          class="MuiFormGroup-root MuiFormGroup-row css-mocked-MuiFormGroup-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium contained-button css-mocked-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Configure RSUs
          </button>
        </div>
      </div>
    </div>
    <div
      class="container-fluid"
      style="width: 100%; height: calc(100vh - 100px); display: flex;"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      />
    </div>
  </div>
</div>
`;
